lib/Dialect/Arc/Transforms/StripSV.cpp

```diff
@@ -0,0 +1,178 @@
+                                     StringAttr::get(&getContext(), "test_en"),
+                                     StringAttr::get(&getContext(), "en")});
+  auto expectedClockGateOutputs =
+      ArrayAttr::get(&getContext(), {StringAttr::get(&getContext(), "out")});
+  auto i1Type = IntegerType::get(&getContext(), 1);
+
+  for (auto extModOp : mlirModule.getOps<hw::HWModuleExternOp>()) {
+    if (extModOp.getVerilogModuleName() == "EICG_wrapper") {
+      if (extModOp.getArgNames() != expectedClockGateInputs ||
+          extModOp.getResultNames() != expectedClockGateOutputs) {
+        extModOp.emitError("clock gate module `")
+            << extModOp.moduleName() << "` has incompatible port names "
+            << extModOp.getArgNamesAttr() << " -> "
+            << extModOp.getResultNamesAttr();
+        return signalPassFailure();
+      }
+      if (extModOp.getArgumentTypes() !=
+              ArrayRef<Type>{i1Type, i1Type, i1Type} ||
+          extModOp.getResultTypes() != ArrayRef<Type>{i1Type}) {
+        extModOp.emitError("clock gate module `")
+            << extModOp.moduleName() << "` has incompatible port types "
+            << extModOp.getArgumentTypes() << " -> "
+            << extModOp.getResultTypes();
+        return signalPassFailure();
+      }
+      clockGateModuleNames.insert(extModOp.moduleNameAttr());
+    } else {
+      externModuleNames.insert(extModOp.moduleNameAttr());
+    }
+    opsToDelete.push_back(extModOp);
+  }
+  LLVM_DEBUG(llvm::dbgs() << "Found " << clockGateModuleNames.size()
+                          << " clock gates, " << externModuleNames.size()
+                          << " other extern modules\n");
+
+  // Remove `sv.*` operation attributes.
+  mlirModule.walk([](Operation *op) {
+    auto isSVAttr = [](NamedAttribute attr) {
+      return attr.getName().getValue().startswith("sv.");
+    };
+    if (llvm::any_of(op->getAttrs(), isSVAttr)) {
+      SmallVector<NamedAttribute> newAttrs;
+      newAttrs.reserve(op->getAttrs().size());
+      for (auto attr : op->getAttrs())
+        if (!isSVAttr(attr))
+          newAttrs.push_back(attr);
+      op->setAttrs(newAttrs);
+    }
+  });
+
+  // Remove ifdefs and verbatim.
+  for (auto verb : mlirModule.getOps<sv::VerbatimOp>())
+    opsToDelete.push_back(verb);
+  for (auto verb : mlirModule.getOps<sv::IfDefOp>())
+    opsToDelete.push_back(verb);
+
+  for (auto module : mlirModule.getOps<hw::HWModuleOp>()) {
+    for (Operation &op : *module.getBodyBlock()) {
+      // Remove ifdefs and verbatim.
+      if (isa<sv::IfDefOp, sv::CoverOp, sv::CoverConcurrentOp>(&op)) {
+        opsToDelete.push_back(&op);
+        continue;
+      }
+      if (isa<sv::VerbatimOp, sv::AlwaysOp>(&op)) {
+        opsToDelete.push_back(&op);
+        continue;
+      }
+
+      // Remove wires.
+      if (auto assign = dyn_cast<sv::AssignOp>(&op)) {
+        auto wire = assign.getDest().getDefiningOp<sv::WireOp>();
```

Comment:
I assume hw.wire is what you need so this doesn't break on fields.

Reply:
Yeah `hw.wire` makes pretty much all of this obsolete. There are some corner cases in the FIRRTL-to-HW lowering where we still see some `sv.wire`s, but those we can just reject until we figure out a non-SV representation (like force/release, assert/assume/cover, etc.).